cli/cmd/metrics.go

```diff
@@ -177,30 +177,16 @@ func getPodsFor(ctx context.Context, clientset kubernetes.Interface, namespace s
 		if err != nil {
 			return nil, err
 		}
-		matchLabels = deployment.Spec.Selector.MatchLabels
-		ownerUID = deployment.GetUID()
+		templateLabels := deployment.Spec.Template.Labels
+		var podList *corev1.PodList
 
-		replicaSets, err := clientset.AppsV1().ReplicaSets(namespace).List(
-			ctx,
-			metav1.ListOptions{
-				LabelSelector: labels.Set(matchLabels).AsSelector().String(),
-			},
-		)
+		podList, err = clientset.CoreV1().Pods(namespace).List(ctx, metav1.ListOptions{
+			LabelSelector: labels.Set(templateLabels).AsSelector().String(),
+		})
 		if err != nil {
 			return nil, err
 		}
-
-		var pods []corev1.Pod
-		for _, rs := range replicaSets.Items {
-			if isOwner(ownerUID, rs.GetOwnerReferences()) {
-				podsRS, err := getPodsFor(ctx, clientset, namespace, fmt.Sprintf("%s/%s", k8s.ReplicaSet, rs.GetName()))
-				if err != nil {
-					return nil, err
-				}
-				pods = append(pods, podsRS...)
-			}
-		}
-		return pods, nil
+		return podList.Items, nil
```

Comment:
This `isOwner` check is in place to ensure we only select the pods owned by the given Deployment. This change reverts a bug fix made in #5567.

To observe the issue:
```bash
$ curl https://run.linkerd.io/emojivoto.yml | linkerd inject - | kubectl apply -f -
$ curl https://run.linkerd.io/emojivoto-statefulset.yml | linkerd inject - | kubectl apply -f -
$ curl https://run.linkerd.io/emojivoto-daemonset.yml | linkerd inject - | kubectl apply -f -

$ bin/go-run cli diagnostics proxy-metrics -n emojivoto deploy/web | grep "# POD"
# POD web-0 (1 of 3)
# POD web-5f86686c4d-c7td9 (2 of 3)
# POD web-6qsh2 (3 of 3)
```